solutions/swb-reference/integration-tests/tests/multiStep/costCenter.test.ts

```diff
@@ -1,18 +1,62 @@
 import ClientSession from '../../support/clientSession';
 import Setup from '../../support/setup';
+import { PaabHelper } from '../../support/complex/paabHelper';
+import { checkHttpError } from '../../support/utils/utilities';
+import HttpError from '../../support/utils/HttpError';
 
 describe('multiStep costCenter test', () => {
   const setup: Setup = Setup.getSetup();
+  const paabHelper = new PaabHelper();
   let adminSession: ClientSession;
+  let pa1Session: ClientSession;
+  let rs1Session: ClientSession;
 
   beforeAll(async () => {
     adminSession = await setup.getDefaultAdminSession();
+    ({ pa1Session, rs1Session } = await paabHelper.createResources());
   });
 
   afterAll(async () => {
     await setup.cleanup();
   });
 
+  async function checkThatRoleCannotAccessCostCenter(
+    role: 'researcher' | 'projectAdmin',
+    costCenterId: string
+  ) {
+    console.log(`Check ${role} cannot access cost center`);
+    let session: ClientSession;
+    if (role === 'researcher') {
+      session = rs1Session;
```

Comment:
not a requirement but you could pass in session as a param with the role to make this a little more functional

Reply:
Ah, I didn't add the session in as a param because I didn't want a mismatch between the session being passed in and the role. Devs could accidentally pass in a PA session while setting a researcher role.